iceoryx_dds/include/iceoryx_dds/internal/gateway/iox_to_dds.inl

```diff
@@ -29,52 +31,57 @@ namespace dds
 // ======================================== Public ======================================== //
 template <typename channel_t, typename gateway_t>
 inline Iceoryx2DDSGateway<channel_t, gateway_t>::Iceoryx2DDSGateway() noexcept
-    : gateway_t()
+    : gateway_t(capro::Interfaces::DDS, DISCOVERY_PERIOD, FORWARDING_PERIOD)
 {
 }
 
 template <typename channel_t, typename gateway_t>
-inline void Iceoryx2DDSGateway<channel_t, gateway_t>::loadConfiguration(const GatewayConfig& config) noexcept
+inline void Iceoryx2DDSGateway<channel_t, gateway_t>::loadConfiguration(const gw::GatewayConfig& config) noexcept
 {
-    iox::LogDebug() << "[Iceoryx2DDSGateway] Configuring gateway.";
+    LogDebug() << "[Iceoryx2DDSGateway] Configuring gateway...";
     for (const auto& service : config.m_configuredServices)
     {
         if (!this->findChannel(service.m_serviceDescription).has_value())
         {
-            setupChannel(service.m_serviceDescription);
+            auto serviceDescription =  service.m_serviceDescription;
+            LogDebug() << "[DDS2IceoryxGateway] Setting up channel for service: {"
+                            << serviceDescription.getServiceIDString() << ", "
```

Comment:
What is the difference then?

Reply:
When we decide to change something on the ServiceDescription, you don't have to change something here, because it will be reflected in the serialization